doc/guides/west/build-flash-debug.rst

```diff
@@ -53,86 +53,124 @@ Building: ``west build``
-(i.e. the folder containing a :file:`CMakeLists.txt` file) of the Zephyr
-application of your choice and running::
+Basics
+======
+
+The easiest way to use it is to go to an application's root directory (i.e. the
+folder containing its :file:`CMakeLists.txt`) and then run::
 
   west build -b <BOARD>
 
 Where ``<BOARD>`` is the name of the board you want to build for. This is
 exactly the same name you would supply to CMake if you were to invoke it with:
 ``cmake -DBOARD=<BOARD>``.
-A build folder named :file:`build` (default name) will be created and the
-build output will be placed in it.
 
-To specify the build directory, use ``--build-dir`` (or ``-d``)::
+.. tip::
 
-  west build -b <BOARD> --build-dir path/to/build/directory
+   You can use the :ref:```west boards`` <west-boards>` command to list all
+   supported boards.
 
-Since the build directory defaults to :file:`build`, if you do not specify
-a build directory but a folder named :file:`build` is present, that will be used,
-allowing you to incrementally build from outside the :file:`build` folder with
-no additional parameters.
+A build directory named :file:`build` will be created, and the application will
+be compiled there after ``west build`` runs CMake to create a build system in
+that directory. If you run ``west build`` with an existing build directory, the
+application is incrementally re-compiled without re-running CMake.
 
-.. note::
-  The ``-b <BOARD>`` parameter is only required when there is no CMake cache
-  present at all, usually when you are building from scratch or creating a
-  brand new build directory. After that you can rebuild (even with additional
-  parameters) without having to specify the board again. If you're unsure
-  whether ``-b`` is required, just try leaving it out. West will print an
-  error if the option is required and was not given.
+You don't need to give ``-b`` if you've already got an existing build
+directory; ``west build`` can figure out the board from the CMake cache. For
+new builds, the :envvar:`BOARD` environment variable and ``build.board``
+configuration option can also be used to set the board.
 
-.. tip::
-  You can use the :ref:```west boards`` <west-boards>` command to list all
-  supported boards.
+To use another build directory, use ``--build-dir`` (or ``-d``)::
 
-Specify the source directory path as the first positional argument::
+  west build -b <BOARD> --build-dir path/to/build/directory
+
+To specify the application source directory explicitly, give its path as a
+positional argument::
 
   west build -b <BOARD> path/to/source/directory
 
-Additionally you can specify the build system target using the ``--target``
-(or ``-t``) option. For example, to run the ``clean`` target::
+To specify the build system target to run, use ``--target`` (or ``-t``). For
+example, to run the ``pristine`` target, which deletes all the files in the
+build directory::
 
-  west build -t clean
+  west build -t pristine
```

Comment:
We should use `-t run` here. There's already `-p` for `pristine` so not the greatest example

Reply:
I'm switching to this on purpose to make the -p switch more clear

Reply:
I've added several different target examples now.